frontend/src/libs/components/prompt-generation/prompt-generation.tsx

```diff
@@ -0,0 +1,48 @@
+import { Button, Icon } from '~/libs/components/components.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useCallback, useState } from '~/libs/hooks/hooks.js';
+
+import { PromptCard } from './libs/components/components.js';
+import { PromptCategory } from './libs/enums/enums.js';
+import styles from './styles.module.scss';
+
+const PromptGeneration: React.FC = () => {
+  const [isSpinning, setIsSpinning] = useState(false);
+
+  const promptCategories = Object.values(PromptCategory);
+
+  const handleButtonClick = useCallback((): void => {
+    setIsSpinning(true);
+
```

Comment:
Create generatedPrompt object and set it up with random mocked data per prompt category - then show up that generatedPrompt on promptCard

Reply:
Do I understand correctly that each time the button is clicked, the values in generatedPrompt object are changed randomly with ones in mocked data?

Reply:
Yes